mixer/adapter/svcctrl/svcctrl_test.go

```diff
@@ -17,49 +17,141 @@ package svcctrl
 import (
 	"reflect"
 	"testing"
-	"time"
 
-	"github.com/davecgh/go-spew/spew"
-	sc "google.golang.org/api/servicecontrol/v1"
-)
+	pbtypes "github.com/gogo/protobuf/types"
 
-func TestHandleMetric(t *testing.T) {
-	timeNow := time.Now().Format(time.RFC3339Nano)
-	request, err := handleMetric(timeNow, "")
+	"istio.io/istio/mixer/adapter/svcctrl/config"
+	"istio.io/istio/mixer/adapter/svcctrl/template/svcctrlreport"
+	at "istio.io/istio/mixer/pkg/adapter/test"
+	"istio.io/istio/mixer/template/apikey"
+	"istio.io/istio/mixer/template/quota"
+)
 
+func TestInitializeHandlerContext(t *testing.T) {
+	adapterCfg := getTestAdapterConfig()
+	ctx, err := initializeHandlerContext(at.NewEnv(t), adapterCfg, &mockSvcctrlClient{})
 	if err != nil {
-		t.Fatalf("handleMetric() failed with:%v", err)
+		t.Errorf("initializeHandlerContext() failed with %v", err)
+	}
+
+	expectedIdx := map[string]*config.GcpServiceSetting{
+		"service_a": adapterCfg.ServiceConfigs[0],
+		"service_b": adapterCfg.ServiceConfigs[1],
+	}
+	if !reflect.DeepEqual(expectedIdx, ctx.serviceConfigIndex) {
+		t.Errorf("expect serviceConfigIndex :%v, but get %v",
+			expectedIdx, ctx.serviceConfigIndex)
+	}
+}
+
+func TestConfigValidation(t *testing.T) {
+
+	{
```

Comment:
why is this floating block here?

Reply:
That is a scope to separate a test case.